include/net/net_mgmt.h

```diff
@@ -140,6 +140,7 @@ struct net_mgmt_event_callback {
 
 #ifdef CONFIG_NET_MGMT_EVENT_INFO
 	const void *info;
+	size_t info_length;
```

Comment:
It was by design: length does not need to be passed since for the event it must be a certain struct for the info. So length is implied by the event code.

Reply:
The design looks a bit weird then. We need to have the length here, how the recipient would know the size otherwise. It is probably ok for IP address but now so much for something more complex.

Reply:
Design was totally fine within net_mgmt api. Since for such event with an info, it is guaranteed to have that structure S etc... (Note as well this was done like this when tiny targets were a priority).

Now, in the current case of having a socket layer on top, that makes sense to have a length since that part is not going to interpret each and every commands to set the length in its own header. That would be indeed tedious. Can you change your commit message to reflect that?

Reply:
> Now, in the current case of having a socket layer on top

Don't blame sockets, blame userspace, that's what not "compatible" with fine optimizations like that. A move to introduce poorman's netlink layer wasn't exactly expected right away, but quite natural reaction to userspace hurdles...

Reply:
Well, in fact, we could have the same thing in AF_NET_MGMT: user buffer length would need to be sized to the biggest possible info size, and up to the user - depending on event - to cast it to right struct. If buffer is too small, error would be returned. But where it is fine in the kernel to have such optimizations, I feel a bit uncomfortable to get the same in a socket api.